server/src/main/java/org/candlepin/model/CertificateSerialCurator.java

```diff
@@ -37,68 +35,96 @@
  */
 public class CertificateSerialCurator extends AbstractHibernateCurator<CertificateSerial> {
 
-    private static int inClauseLimit = 1000;
-
     public CertificateSerialCurator() {
         super(CertificateSerial.class);
     }
 
     /**
-     * @return list of certificate serials which are revoked but not yet collected
-     * and put into CRL
+     * Fetches a collection of serials from uncollected, revoked certficiate serials. If there are
+     * no such certificate serials, this method returns an empty collection.
+     *
+     * @return
+     *  a collection of serials from uncollected, revoked certificate serials
      */
-    @SuppressWarnings("unchecked")
-    public CandlepinQuery<CertificateSerial> retrieveTobeCollectedSerials() {
+    public CandlepinQuery<Long> getUncollectedRevokedCertSerials() {
         DetachedCriteria criteria = DetachedCriteria.forClass(CertificateSerial.class)
             .add(Restrictions.eq("revoked", true))
-            .add(Restrictions.eq("collected", false));
+            .add(Restrictions.eq("collected", false))
+            .setProjection(Projections.id()); // Note: the ID *is* the serial for cert serials
 
-        return this.cpQueryFactory.<CertificateSerial>buildQuery(this.currentSession(), criteria);
+        return this.cpQueryFactory.<Long>buildQuery(this.currentSession(), criteria);
     }
 
-    @SuppressWarnings("unchecked")
-    public CandlepinQuery<CertificateSerial> getExpiredSerials() {
-        //TODO - Should date fields be truncated when checking expiration?
-
+    /**
+     * Fetches a collection of serials from revoked certficiate serials that expired prior to
+     * "today," where "today" is defined as "00:00:00 of the current local date on the system
+     * running the Candlepin instance in which this method is called." If there are no such
+     * certificate serials, this method returns an empty collection.
+     *
+     * @return
+     *  a collection of serials from revoked certficiate serials that expired prior to "today."
+     */
+    public CandlepinQuery<Long> getExpiredRevokedCertSerials() {
         DetachedCriteria criteria = DetachedCriteria.forClass(CertificateSerial.class)
-            .add(Restrictions.le("expiration", Util.yesterday()))
```

Comment:
I'm not sure I follow the argument here. No matter what time zone we're using, the comparison is still being done as a pair of longs relative to the epoch. The change here to "midnight" changes the cutoff from a moving window to a fixed value; this would give us the option of reducing the frequency the CRL is resync'd. In either case, when this method is called, it's only going to pull serials which have expired from the view of CP.

I'm content with moving it back further to midnight-yesterday, but I feel focusing on a date resolution rather than millisecond resolution is cleaner here, and given that there was a comment about something similar, I assume this was debated in the past.